pkg/controller/deployment/deployment_controller.go

```diff
@@ -341,6 +397,11 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 		return nil
 	}
 
+	err = dc.classifyReplicaSets(deployment)
+	if err != nil {
+		glog.Infof("Classifying ReplicaSets failed")
+	}
+
 	if d.DeletionTimestamp != nil {
```

Comment:
Also the deletionTimestamp check needs to be after the overlap check (otherwise we update the status for overlapping deployments once they are marked deleted.